### ISSUE TITLE
Bitwarden Autofill in browser sometimes break keyboard's functionality

### ISSUE THREAD

**Issue by awooooool** [bitwarden/browser#1535]
## Describe the Bug

On rare occasion, after triggering autofill from Bitwarden extension, I can't type with my keyboard anymore. Shortcuts like Win + S to open windows search and media button still works fine.
## Steps To Reproduce

1. Go to any site that has login page
2. Autofill using extension
3. Username and password filled just fine
4. Can't type anymore on random occasion

## Expected Result

Keyboard will work just fine even after using autofill
## Actual Result
Keyboard stops working

## Screenshots or Videos

N/A
## Environment

- Operating system: Windows 10 Enterprise 2004 19041.746 (64-bit)
- Browser: Firefox Developer Edition 85.0b9 (64-bit)
- Build Version (go to "Settings" → "About" in the app): 1.47.1

## Additional Context

N/A

**Comment by wusatosi** [comment]
This doesn't happen to me and has not happened to me. Can you present a more reproducible case or record a video of this occurring.

**Comment by awooooool** [comment]
> 
> 
> This doesn't happen to me and has not happened to me. Can you present a more reproducible case or record a video of this occurring.

It's random. Even I'm confused on how to replicate this one. It's like I keep using autofill in browser until it somehow breaks keybuard functionality. Copying and pasting from extensions work just fine though.

**Comment by vfandrea** [comment]
I have the same issue. Sometimes after filling a form with Bitwarden for Chrome I can't type anymore with my keyboard. "Special keys" (like Windows key) are working, but numbers and letters are not working anymore with both keyboard (I use a laptop with a wireless keyboard).

The only solution is to reboot the computer.

**Comment by willfliaw** [comment]
Some people (myself included) have also commented about [this issue](https://www.reddit.com/r/Bitwarden/comments/mxg1om/using_firefox_browser_extenstion_stops_keyboard/?utm_source=share&utm_medium=web2x&context=3) on the past few weeks on the Bitwarden subreddit. Still, despite being affected by said issue, I've not been able do determine a common denominator so its occurrence seems to be "random" (occurring only after the mentioned steps to reproduce).

**Comment by awooooool** [comment]
Idk if it's just placebo effect or what, but it seems that chances for freezing the keyboard are reduced(?) if you use the context menu for filling instead from the extension menu. I'm not really sure if that's really the case since I've also updated my Firefox Dev to 89.0b7.

**Comment by pat22296** [comment]
I have exactly the same issue on 2 of my PCs. There seems to be no reliable way to reproduce it. Somedays it doesnt happen at all but at other days it happens 3 times in a row. The only fix for me seems to be a reebot. I would really appreciate it if you could fix this annoying issue...

**Comment by willfliaw** [comment]
As reddit user u/faberffm said in this [thread ](https://www.reddit.com/r/Bitwarden/comments/mxg1om/using_firefox_browser_extenstion_stops_keyboard/?utm_source=share&utm_medium=web2x&context=3) there seems to be an unresolved conflict with Bitwarden and Karperky's browser extension. Disabling the latter solved the issue for me (at least I haven't been through the issue in the past week).

**Comment by awooooool** [comment]
That makes sense, I also use Kaspersky on my laptop. Also I found out that if I use autofill again after my keyboard stops responding, my keyboard works again. I just tried that once last night.

**Comment by wusatosi** [comment]
> That makes sense, I also use Kaspersky on my laptop. Also I found out that if I use autofill again after my keyboard stops responding, my keyboard works again. I just tried that once last night.

This happens to me too, but I don't have Kaspersky installed.

**Comment by rcjkierkels** [comment]
Same issue here. I do have Kaspersky.

**Comment by vfandrea** [comment]
Did someone already try to disable the virtual on-screen keyboard from Kaspersky?

I'm testing this workaround since 2 weeks and it seems to work, but considering this issue is random I can't confirm at this moment (more testing is needed).

**Comment by awooooool** [comment]
> 
> 
> Did someone already try to disable the virtual on-screen keyboard from Kaspersky?
> 
> I'm testing this workaround since 2 weeks and it seems to work, but considering this issue is random I can't confirm at this moment (more testing is needed).

where can I disable it?

**Comment by vfandrea** [comment]
Go to Kaspersky Settings Window then select "Advanced". Here you will find the setting for on screen keyboard.

More details here: [Kaspersky Internet Security  - How to configure the display of the On-Screen Keyboard icon](https://support.kaspersky.com/KIS/21.3/en-US/90458.htm)

**Comment by awooooool** [comment]
> 
> 
> Go to Kaspersky Settings Window then select "Advanced". Here you will find the setting for on screen keyboard.
> 
> More details here: [Kaspersky Internet Security - How to configure the display of the On-Screen Keyboard icon](https://support.kaspersky.com/KIS/21.3/en-US/90458.htm)
> 
> I confirm that since i deactivated the virtual keyboard the issue didn't come back.

I disabled "Secure Keyboard Input" for password fields instead of "On-Screen Keyboard" 2 days ago. Haven't see the issue since then.

**Comment by vfandrea** [comment]
Finally I can confirm that disabling the virtual Keyboard in Kaspersky the issue has disappeared.

**Comment by Whaouu** [comment]
Hello,

I also regularly have this problem on 2 machines using Bitwarder and Kaspersky Internet Security.

It is very likely that there is an interaction problem between these 2 plug-ins.

I should mention that I have been using LastPass and Kaspersky plug-ins for many years without any problem, the partial keyboard blocking has appeared since I replaced LastPass with Bitwarden.

It doesn't seem right to have to disable a Kaspersky feature to fix a Bitwarden plug-in problem...

I also found a method to unlock the keyboard. Just open a new tab in Firefox, then press Home to open the Google search page (I have it set as Home), then type some letters in the search field.

**Comment by jophut** [comment]
Hi there,

I can confirm the same erroneous behaviour when using the Bitwarden Chrome Extensions together with the Kaspersky Chrome Extensions.
Disabling the "On-Screen Keyboard" did not work out.
Disabling the Kaspersky Chrome Extension solved my problem.

A different work-around:
Use Bitwarden to fill the username&password input fields, but before clicking the "Login-button" make sure a non-password field is active, e.g. click the username field.
It looks like the Kaspersky Extension is only "active" when a password field is highlighted.

**Comment by awooooool** [comment]
> 
> 
> Hi there,
> 
> I can confirm the same erroneous behaviour when using the Bitwarden Chrome Extensions together with the Kaspersky Chrome Extensions.
> Disabling the "On-Screen Keyboard" did not work out.
> Disabling the Kaspersky Chrome Extension solved my problem.
> 
> A different work-around:
> Use Bitwarden to fill the username&password input fields, but before clicking the "Login-button" make sure a non-password field is active, e.g. click the username field.
> It looks like the Kaspersky Extension is only "active" when a password field is highlighted.

I can confirm that after disabling both On-Screen Keyboard and Secure Keyboard Input in Kaspersky, the problem that I had with Bitwarden disappears. I've tested this for 2 weeks or so.

**Comment by eliykat** [comment]
I'm not familiar with Kaspersky, but their [Secure Keyboard Input](https://support.kaspersky.com/KFPWin/6.0/en-EN/74444.htm) feature seems similar to the MacOS secure input that was causing us problems on desktop [here](https://github.com/bitwarden/desktop/issues/298). There might be something about our auto-fill script that is "holding open" Kaspersky's secure input detection. I'm not sure why that would break keyboard input entirely though.

**Comment by awooooool** [comment]
Last night, my keyboard crashed again for the first time since I turned off Kaspersky's OSK and Secure Input. My keyboard won't work anywhere **except** the last webpage that I've used Bitwarden's autofill.

**Comment by luckybhumkar** [comment]
**BitWarden + Kaspersky Extension = Keyboard stops** functioning sometime. Moreover, it starts again if start typing in the search field of most websites including Google. 

Looks strange. I was facing this issue more when LastPass tighten the rule & installed BitWarden.

Now, I have disabled **Secure Keyboard Input** from _Kaspersky Internet Security ➔ Settings ➔ Additionals_ 

Hopefully, it may solve my keyboard problem. I will update after 1 month. By the way big thanks to contributors to this thread.

**Comment by awooooool** [comment]
I think this and Kaspersky problem is even more frequent when trying to log into AWS Console

**Comment by luckybhumkar** [comment]
Absolutely @awooooool. Never faced a keyboard/input issue after disabling the **Secure Keyboard Input** tool.

**Comment by piemaniac** [comment]
I have also been having this issue. Also have Kaspersky installed.

I even replaced a keyboard as I thought it might have been the keyboard originally. Like others, I eventually noticed I could get the keyboard to work after typing in a search field on a website (facebook, google etc). That's served as my fix until now.

I received a new laptop yesterday and setup everything as I have on my desktop PC. But this morning the keyboard on the laptop stopped working. I immediately realised the issue has to be do with software I had installed. The first thing I thought of was the Kaspersky add on in Firefox, so I disabled it. I also turned off the secure keyboard in Kaspersky.

I'm hoping this will now resolve the issue.

Like others, this issue came up after switching to Bitwarden from Lastpass. Hopefully someone can work out what the issue is, so that we don't have to utilise work arounds.
